src/main/java/ca/utoronto/fitbook/application/service/PostCreationService.java

```diff
@@ -0,0 +1,70 @@
+package ca.utoronto.fitbook.application.service;
+
+import ca.utoronto.fitbook.application.port.in.*;
+import ca.utoronto.fitbook.application.port.in.command.PostCreationCommand;
+import ca.utoronto.fitbook.application.port.out.SavePostPort;
+import ca.utoronto.fitbook.application.port.out.SaveUserPort;
+import ca.utoronto.fitbook.application.port.out.response.PostCreationResponse;
+import ca.utoronto.fitbook.entity.Post;
+import ca.utoronto.fitbook.entity.User;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.Date;
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class PostCreationService implements PostCreationUseCase{
+    private final SavePostPort savePostPort;
+    private final SaveUserPort saveUserPort;
+    private final LoadUserPort loadUserPort;
+    private final CheckUserExistsPort checkUserExistsPort;
+    private final LoadExerciseListPort loadExerciseListPort;
+
+    @Override
+    public PostCreationResponse createPost(PostCreationCommand command) {
+        // Certify user existence
+        String userId = command.getUserId();
+        if (!checkUserExistsPort.checkUserExists(userId))
+            throw new UserNotFoundException("User does not exist");
```

Comment:
I could be misunderstanding the code, but I think the userId should be passed when constructing and throwing the UserNotFoundException
```suggestion
            throw new UserNotFoundException(userId);
```